### PR TITLE
Pruning Tests and Bug Fixes

### PR DESCRIPTION
**Pull request by AlexandraRoatis**
## Type of change

- [x] Bug fix.
- [ ] New feature.
- [x] Enhancement.
- [x] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

### CODE REVIEW

File: modAionImpl/src/org/aion/zero/impl/db/AionRepositoryImpl.java [hunk 1]
```diff
@@ -512,10 +512,6 @@ public void flush() {
                         db.commit();
                     }
                 }
-            } else {
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/db/AionRepositoryImpl.java]: What scenario we will see the databaseGroup equal to null ? Sometimes We can see this message during the seednode running, also the project test case.

Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/zero/impl/db/AionRepositoryImpl.java]: The databaseGroup is null when using a repository snapshot. The snapshot is always used when importing a side chain block. The message is really useless, cause it doesn't notify us of anything meaningful.

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionHub.java [hunk 2]
```diff
@@ -427,14 +427,14 @@ public BlockContext getNewMiningBlockTemplate(BlockContext oldBlockTemplate, lon
...
+        Block best = blockchain.getBestBlock();
+        if (best.getHeader().getSealType() == BlockHeader.BlockSealType.SEAL_POS_BLOCK) {
             return null;
         } else {
             StakingBlock blockTemplate;
             blockTemplateLock.lock();
             try {
-                blockTemplate = blockchain.createStakingBlockTemplate(
-                        mempool.getPendingTransactions(), signingPublicKey, newSeed, coinbase);
+                blockTemplate = blockchain.createStakingBlockTemplate(best, mempool.getPendingTransactions(), signingPublicKey, newSeed, coinbase);
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/blockchain/AionHub.java]: Should we just use blockTemplateLock and the try...finally to include all of the method?
Same as the getMiningBlockTemplate

Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/zero/impl/blockchain/AionHub.java]: I added an extra commit for this change.

**Pull request merged**